### PR TITLE
[Offload][NFC] Add offload subfolder and README

### PR DESCRIPTION
**Pull request by jdoerfert**
The readme only states the goal and has links to further information, e.g., our meetings.

### CODE REVIEW

File: offload/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+users to execute code on accelerators or other "co-processors" that may or may
+not match the architecture of their "host". In the long run, all kinds of
+targets are in scope of this effort, including but not limited to: CPUs, GPUs,
+FPGAs, AI/ML accelerators, distributed resources, etc.
+
+The project is just starting and the design is still not ironed out. More
+content will show up here and on our webpage soon. In the meantime people are
+encouraged to participate in our meetings (see below) and check our
+[development board](https://github.com/orgs/llvm/projects/24/) as well as the
+discussions on [discord](https://discourse.llvm.org/tag/offload).
```
Comment (shiltian) [inline offload/README.md:13]:
```suggestion
discussions on [Discourse](https://discourse.llvm.org/tag/offload).
```

Comment (shiltian) [reply offload/README.md:13]: Unless you also opened a channel at Discord.

File: offload/README.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+# Meetings
+
+Every second Friday, starting Jan 12, 2024, 9:00 - 10:00am PST.
+[invite.ics](https://docs.google.com/document/d/1PAeEshxHCv22JDBCPA9GXGggLp0t7rsnD_jL04MBbzw/edit?usp=sharing)
```
Comment (shiltian) [inline offload/README.md:18]: This URL is the doc.
```suggestion
[invite.ics](https://drive.google.com/file/d/10Gv-MuEQ-VztJBqpQBggJclv2vU5VF0V/view?usp=sharing)
```

**Review by shiltian — APPROVED** [review]
LGTM. I'm not sure if we need special approval to add a folder at top level.

**Review by jplehr — APPROVED** [review]
LGTM

File: offload/README.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+The project is just starting and the design is still not ironed out. More
+content will show up here and on our webpage soon. In the meantime people are
+encouraged to participate in our meetings (see below) and check our
+[development board](https://github.com/orgs/llvm/projects/24/) as well as the
+discussions on [Discourse](https://discourse.llvm.org/tag/offload).
+
+# Meetings
+
+Every second Friday, starting Jan 12, 2024, 9:00 - 10:00am PST.
```
Comment (saiislam) [inline offload/README.md:17]: Meeting time and ics file needs to be updated before landing this PR.

**Review by jhuber6 — APPROVED** [review]

**Pull request merged**